native/src/routes/JpalTracking.tsx

```diff
@@ -55,6 +55,12 @@ const JpalTracking = ({ navigation, route }: PropsType): ReactElement => {
     updateTrackingEnabled(!trackingEnabled)
   }
 
+  useEffect(() => {
```

Comment:
Can you elaborate please? The example shown there is a little bit different because it is only about internal `useState` state while we have persisted async storage state. I'll try to avoid the `useEffect` but maybe you can help here?

Reply:
@SteffiStuffel the correct way would probably be to update the async storage in `navigateToDeepLink` I guess and just load the async storage in `JpalTracking` without updating anything there (with the exception of after user actions of course). What do you think?

Reply:
I thought we would need something like 

```
  const disablingChanged = disableTracking === trackingEnabled
  if (disablingChanged && disableTracking) {
    updateTrackingEnabled(false)
  }
```

Problem here is not saving in the async storage, but that both can update:
1) the disableTracking from the route prop
2) the trackingEndabled by clicking the toggle

So we cannot - like in the example in the link - check if prop updated, by comparing it with the "old value"=trackingEnabled, as this value can also change. 

So we need (like in the link) introduce a new variable and then check for equality.
This should work:

```
  const [oldDisableTracking, setOldDisableTracking] = useState<boolean | undefined>(undefined)
  if (disableTracking !== undefined && disableTracking !== oldDisableTracking) {
    setOldDisableTracking(disableTracking)
    updateTrackingEnabled(false)
  }
```

Reply:
But in this case, maybe it is the better solution to stick to the useEffect.

Reply:
What do you think about the proposal from above (updating async storage in navigateToDeepLink)?

Reply:
I like it, as it is better architectural design.